src/file_pipe_log/pipe_builder.rs

```diff
@@ -39,6 +67,104 @@ pub struct DualPipesBuilder<B: FileBuilder> {
 
     append_files: Vec<FileToRecover>,
     rewrite_files: Vec<FileToRecover>,
+    extra_params: Option<TruncateQueueParameter>,
+}
+
+trait RecoverQueue {
```

Comment:
Is it necessary? Trait only exists to help us write generic code, I don't see any use of it currently.

Reply:
No necessary, As you have commented, it's advise to add a `recover_queue` trait, so i do. before I push the commit, I have noticed that the trait have not used. so i ask for your help when i pushed the commit

Reply:
Oh, I didn't mean trait by "interface". A method is enough.